docs/fundamentals/networking/http/httpclient-migrate-from-httpwebrequest.md

```diff
@@ -0,0 +1,640 @@
+
+#### Example: Set Custom Request Headers
+
+**Setting Default Custom Request Headers in HttpClient**
+
+```csharp
+var client = new HttpClient();
+client.DefaultRequestHeaders.Add("Custom-Header", "value");
+```
+
+**Setting Custom Request Headers in HttpRequestMessage**
+
+```csharp
+var request = new HttpRequestMessage(HttpMethod.Get, uri);
+request.Headers.Add("Custom-Header", "value");
+```
+
+#### Example: Set Common Request Headers
+
+When working with `HttpRequestMessage` in .NET, setting common request headers is essential for providing additional information to the server about the request being made. These headers can include authentication tokens, content types and more. Properly configuring these headers ensures that your HTTP requests are processed correctly by the server.
+For a comprehensive list of common properties available in HttpRequestHeaders, you can refer to the [official documentation](https://learn.microsoft.com/dotnet/api/system.net.http.headers.httprequestheaders#properties).
+
+To set common request headers in `HttpRequestMessage`, you can use the `Headers` property of the `HttpRequestMessage` object. This property provides access to the `HttpRequestHeaders` collection, where you can add or modify headers as needed.
+
+**Setting Common Default Request Headers in HttpClient**
+
+```csharp
+using System.Net.Http.Headers;
+
+var client = new HttpClient();
+client.DefaultRequestHeaders.Authorization =  new AuthenticationHeaderValue("Bearer", "token");
+```
+
+**Setting Common Request Headers in HttpRequestMessage**
+
+```csharp
+using System.Net.Http.Headers;
+
+var request = new HttpRequestMessage(HttpMethod.Get, uri);
+request.Headers.Authorization = new AuthenticationHeaderValue("Bearer", "token");
+```
+
+### Example: Set Content Headers
+
+Content headers are used to provide additional information about the body of an HTTP request or response. When working with `HttpClient` in .NET, you can set content headers to specify the media type, encoding, and other metadata related to the content being sent or received. Properly configuring content headers ensures that the server and client can correctly interpret and process the content.
+
+```csharp
+var client = new HttpClient();
+var request = new HttpRequestMessage(HttpMethod.Post, uri);
+
+// Create the content and set the content headers
+var jsonData = "{\"key\":\"value\"}";
+var content = new StringContent(jsonData, Encoding.UTF8, "application/json");
+// This is also corresponding example to set `MediaType` in `HttpWebRequest`.
+content.Headers.ContentType = new MediaTypeHeaderValue("application/json");
+content.Headers.ContentEncoding.Add("utf-8");
+// WARNING: This is not needed in most of the cases, because most `HttpContent` implementations will calculate this field automatically.
+content.Headers.ContentLength = jsonData.Length;
```

Comment:
```suggestion

// The following headers are set automatically by `StringContent`. If you wish to override their values, you can do it like so:
// content.Headers.ContentType = new MediaTypeHeaderValue("application/json; charset=utf-8");
// content.Headers.ContentLength = Encoding.UTF8.GetByteCount(jsonData);
```
`StringContent` already sets the content-type for you, also accounting for the encoding used.
Despite the name, `ContentEncoding` isn't used to communicate the type of text encoding used.